src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -52,69 +52,427 @@ public class RepoConfiguration {
+         *
+         * @param isIgnoreGlobListOverriding Checks if the list of ignored glob is overridden.
+         * @return This builder object.
+         */
+        public RepoBuilder isIgnoreGlobListOverriding(boolean isIgnoreGlobListOverriding) {
+            this.isIgnoreGlobListOverriding = isIgnoreGlobListOverriding;
+            return this;
+        }
+
+        /**
+         * Updates the {@code isIgnoreCommitListOverriding} for {@code RepoConfiguration}.
+         *
+         * @param isIgnoreCommitListOverriding Checks if the list of ignored commits is overridden.
+         * @return This builder object.
+         */
+        public RepoBuilder isIgnoreCommitListOverriding(boolean isIgnoreCommitListOverriding) {
+            this.isIgnoreCommitListOverriding = isIgnoreCommitListOverriding;
+            return this;
+        }
+
+        /**
+         * Updates the {@code isFileSizeLimitOverriding} for {@code RepoConfiguration}.
+         *
+         * @param isIgnoredAuthorsListOverriding Checks if the list of ignored authors is overridden.
+         * @return This builder object.
+         */
+        public RepoBuilder isIgnoredAuthorsListOverriding(boolean isIgnoredAuthorsListOverriding) {
+            this.isIgnoredAuthorsListOverriding = isIgnoredAuthorsListOverriding;
+            return this;
+        }
+
+        /**
+         * Updates the {@code fileSizeLimit} for {@code RepoConfiguration}.
+         *
+         * @param fileSizeLimit File size limit of the repository.
+         * @return This builder object.
+         */
+        public RepoBuilder fileSizeLimit(long fileSizeLimit) {
+            this.fileSizeLimit = fileSizeLimit;
+            return this;
+        }
+
+        /**
+         * Updates the {@code isFileSizeLimitOverriding} for {@code RepoConfiguration}.
+         *
+         * @param isFileSizeLimitOverriding Checks if the file size limit is overridden.
+         * @return This builder object.
+         */
+        public RepoBuilder isFileSizeLimitOverriding(boolean isFileSizeLimitOverriding) {
+            this.isFileSizeLimitOverriding = isFileSizeLimitOverriding;
+            return this;
+        }
+
+        /**
+         * Updates the {@code isIgnoredFileAnalysisSkipped} for {@code RepoConfiguration}.
+         *
+         * @param isIgnoredFileAnalysisSkipped Checks if the analysis of ignored files is skipped.
+         * @return This builder object.
+         */
+        public RepoBuilder isIgnoredFileAnalysisSkipped(boolean isIgnoredFileAnalysisSkipped) {
+            this.isIgnoredFileAnalysisSkipped = isIgnoredFileAnalysisSkipped;
+            return this;
+        }
+
+        /**
+         * Builds the {@code RepoConfiguration} object with the necessary configurations.
+         *
+         * @return {@code RepoConfiguration}.
+         */
+        public RepoConfiguration build() {
+            return new RepoConfiguration(this);
```

Comment:
should minimally add an assertion to check that location has been set. in general, for all previously disallowed constructor combinations, we want to continue to disallow them (unless we've checked that it doesn't result in undefined/unexpected behavior).

Reply:
I understand, I will look into creating some methods to check the validity of the constructed `RepoConfiguration` object, using the past constructors as a rough guideline on the correct constructor combinations to allow!

Reply:
@gok99 Can I also check if it is fine to set default values for some of the parameters? I realised from previous code that the constructors chain their calls and set default values for certain fields if they are not provided. I was wondering if it is ok to leave it as it is for now, and check that `RepoLocation` is minimally provided by the user?

The current checks that I had implemented relate to the parameters used in the constructors of the old code, which might not be all too helpful especially when default values for some fields are set (hence they will always be populated and not null).

Reply:
Yes, my concern was that in setting booleans for example, we might be violating downstream assumptions about what combinations of fields were previously not possible (disallowed by the constructors), but are now possible.

Unfortunately since these assumptions are mostly implicit, we can't know for sure, without tracing through the whole pipeline, so I think we can do with just the location check for now. Minimally, existing Builder invocations should still work, and if fields interfere undesirably, we can fix them incrementally.

Reply:
I see, I have implemented a simple check to see if `RepoLocation` has been defined during the build process of the `RepoConfiguration` object. Hope that this temporarily provides some safeguards against illegal combinations of parameters.